src/rpc/mod.rs

```diff
@@ -69,7 +62,9 @@ where
         request_body: ValidationRequestBody,
     ) -> RpcResult<()> {
         let request = ValidationRequest::new(request_body, self.provider());
-        request.validate().await
+        tokio::task::spawn_blocking(move || request.validate())
```

Comment:
In the [tokio documentation](https://docs.rs/tokio/latest/tokio/index.html#cpu-bound-tasks-and-blocking-code) it says:

> If your code is CPU-bound and you wish to limit the number of threads used to run it, you should use a separate thread pool dedicated to CPU bound tasks. For example, you could consider using the [rayon](https://docs.rs/rayon) library for CPU-bound tasks.

@mattsse Is this at all applicable relevant here ?
i.e. do you have any insights / hunches on wether using a thread pool optimized for CPU bound tasks would be possible / beneficial ?

Reply:
yeah, the state root part is a mix of IO and CPU but more dominated by CPU.
so I think ideally this should be performed on a rayon pool, that's also how we handle tracing requests in reth, but I don't have numbers for tokio blocking pool vs rayon pool

Reply:
Thanks, for the feedback.
Created an issue for this: https://github.com/ultrasoundmoney/reth-payload-validator/issues/25